docs/reference/commandline/docker.md

```diff
@@ -46,7 +63,60 @@ variables used by the Go runtime. In particular, you may find these useful:
 See the [Go specification](https://pkg.go.dev/golang.org/x/net/http/httpproxy#Config)
 for details on these variables.
 
-## Configuration files
+### Option types
+
+Single character command line options can be combined, so rather than
+typing `docker run -i -t --name test busybox sh`,
+you can write `docker run -it --name test busybox sh`.
+
+#### Boolean
+
+Boolean options take the form `-d=false`. The value you see in the help text is
+the default value which is set if you do **not** specify that flag. If you
+specify a Boolean flag without a value, this will set the flag to `true`,
+irrespective of the default value.
+
+For example, running `docker run -d` will set the value to `true`, so your
+container **will** run in "detached" mode, in the background.
+
+Options which default to `true` (e.g., `docker build --rm=true`) can only be
+set to the non-default value by explicitly setting them to `false`:
+
+```console
+$ docker build --rm=false .
+```
+
+#### Multi
+
+You can specify options like `-a=[]` multiple times in a single command line,
+for example in these commands:
+
+```console
+$ docker run -a stdin -a stdout -i -t ubuntu /bin/bash
+
+$ docker run -a stdin -a stdout -a stderr ubuntu /bin/ls
```

Comment:
Not for this PR, but we need to look at some of these; `-a` / `--attach` is shown as `list`, and those can take multiple values, but for other flags, Cobra is good at silently ignoring duplicates, and in that case _overriding_ previous values. (e.g. `-i -i=false -i=true -i`)

Also for a follow-up, we should pick a different example than `-a`, as it's really more of an "expert" use-case to set that. Perhaps something like `-e` / `--env` or something common could be a good one to pick as example (the `--env` one can be nice as we could _show_ the result;



```bash
$ docker run --rm -e ENV_ONE=one -e ENV_TWO=two alpine printenv
ENV_ONE=one
ENV_TWO=two
<...>
```.